src/rt/aaA.d

```diff
@@ -152,6 +152,21 @@ private:
         deleted = 0;
         GC.free(obuckets.ptr); // safe to free b/c impossible to reference
     }
+
+    void clear() pure nothrow
+    {
+        foreach (ref b; buckets[firstUsed .. $])
+        {
+            if (b.filled)
+            {
+                b.hash = HASH_DELETED;
+                b.entry = null;
+            }
+        }
+        deleted = used;
+        firstUsed = cast(uint) dim;
+        resize(INIT_NUM_BUCKETS);
```

Comment:
Well, that's definitely not a good reason to use clear. How would you even do that? You can't reserve any bucket data, and it would be silly to add a whole bunch of elements and then clear so you could reserve the buckets. A whim of the implementation might just wreck that "reserve" that you did later on.

I'm not opposed to leaving the array intact, as it doesn't necessarily hurt to keep the memory. But not for your reasoning.

Also keep in mind that if one had a very very large AA, and you called clear on it, you may expect it to relinquish as much memory as possible.